Closes #8: use slack-api everywhere; Closes #12: Moved to Mongo; Fixes #13, Fixes #14:  Improved views handling.

nraomobiquityinc opened the pull request:
- Added express-handlebars
- Added a global layout for all views
- Added richer views for welcome and errors
- Changed authSuccess page to replace the queue div using jQuery rather
  than making a post request via form submit.
- Migrated from redis to Mongo
- Changed all Slack API calls to use `slack-api`.

nraomobiquityinc committed:
Fixes #13, Fixes #14: Improved views handling.
* Added express-handlebars
* Added a global layout for all views
* Added richer views for welcome and errors
* Changed authSuccess page to replace the queue div using jQuery rather
  than making a post request via form submit.

nraomobiquityinc committed:
Closes #12. Swapped redis out for Mongo.

nraomobiquityinc committed:
Beautifying files.

nraomobiquityinc committed:
Closes #8: Changed all requests to Slack to use slack-api.

Ustice commented on authenticationHelper.js:
```diff
@@ -124,156 +85,106 @@ exports.handleOauthCallback = function(request, response) {
+        var userId = res.user_id;
+        var userName = res.user;
+        if (userId === expectedUserId) {
+          dao.saveAccessTokenForUser(accessToken, userId, function(res) {
```
This would be a good place to promisify this function. Then you can simply chain the promises and do all of the error handling at the end.

nraomobiquityinc replied:
Unfortunately, Bluebird Promisify doesn't play well with lodash currying.

Ustice commented on master.js:
```diff
@@ -14,7 +14,7 @@ if (cluster.isMaster) {
 
     log.info('application started')
-    console.log('application started on port:', process.env.PORT || config.port)
+    console.log('application started on port:', process.env.NODE_PORT || config.ports.nodePort)
```
This needs to be process.env.PORT on heroku.

Ustice replied:
Also, update the sample-config.json to reflect the changes.

Ustice commented on models/models.js:
```diff
@@ -0,0 +1,123 @@
+var Schema = mongoose.Schema;
+
+var mongoPort = process.env.MONGO_PORT || config.ports.mongoPort;
+mongoose.connect('mongodb://localhost:' + mongoPort + '/slacker', function(err, res) {
```
This should be parameterized so that we can use [MongoHQ](https://bridge.mongohq.com/signup).

nraomobiquityinc committed:
Code changes after code review:
* Mongo connection string is now defined via configuration files
* Updated sample-config.json to include the mongo connection string
* Reverted changes to config.ports to use "PORT" which is what is
  available on heroku.
* bugfix in dao.js: added guards to make sure we don't look for fields
  on undefined values
* bugfix in authenticationHelper.js: forgot to update call to
  bot.sendMessage which was changed to use Promisified slack-api

nraomobiquityinc committed:
Removing bluebird from package.json as it's not used.
